### PR TITLE
Refactor instrument plugin protocol; add PDS3 label support and READ_FILE_KWARGS forwarding

### PR DESCRIPTION
**Pull request by markshowalter**
## Summary

- Replaces the old per-instrument `detect_vicar`/`detect_fits` split with a unified `read_file(filename, obj, **kwargs)` entry point in each instrument module. The reader cascade in `io.py` now calls each instrument's `read_file` in priority order and returns on the first match, eliminating the separate VICAR and FITS instrument lists.

- Extracts shared format helpers (`try_open_vicar`, `is_fits_file`, `extract_fits_array`, `read_pds3_image_array`) into a new `instruments/_shared.py` module, and moves shared type definitions (`FilterInfo`, `ObjectSelector`, `ReadResult`) into a new `picmaker/_types.py`, resolving the circular-import risk that previously kept them in `io.py`.

- Adds PDS3 `.LBL` label support throughout: `read_one_image_array` now accepts a `pdsparser.PdsLabel` directly or auto-parses paths ending in `.LBL`/`.lbl`, dispatching first through the per-instrument cascade and then through a generic fallback.

- Introduces `READ_FILE_KWARGS` in `options.py` — a tuple of `PicmakerOptions` field names that `_process_one_image` forwards generically to every `read_file` call, so instrument-specific options can be added without touching `io.py` or the pipeline loop. Currently carries `mosaic` and `pds3_label_method`.

- Renames the instrument modules to include the sub-instrument suffix (`cassini.py` → `cassini_iss.py`, `voyager.py` → `voyager_iss.py`, `galileo.py` → `galileo_ssi.py`, `nh.py` → `nh_lorri.py`) and renames `ReadTiff16`/`WriteTiff16` to `read_tiff16`/`write_tiff16` per naming standards.

- Renames the `--hst` CLI flag to `--mosaic` and moves HST mosaic assembly logic into `instruments/hst.py` as `apply_mosaic`, which the pipeline calls via `hasattr` check.

- Updates the developer guide throughout to reflect the new protocol, adds a corrected "Adding an instrument-specific option" section (the previous version omitted two of the four required file edits), and adds `CLAUDE.md`.

- Adds branch-coverage tests across `_filters`, `instruments/__init__`, `instruments/_shared`, `instruments/galileo_ssi`, and `io` to bring every source file to ≥90% coverage, including tests for the PdsLabel-passed-directly path, generic VICAR/FITS fallbacks, sibling `.lbl`/`.LBL` detection, and `get_outfile` edge cases.

- Changes the default `pds3_label_method` from `'strict'` to `'fast'` in `PicmakerOptions`, `images_to_pics`, `read_pds_labeled_image_array`, and all internal `kwargs.get` fallbacks.

## Test plan

- [ ] `bash scripts/run-all-checks.sh` passes (ruff, mypy strict, bandit, vulture, pytest with coverage ≥ 90%)
- [ ] Snapshot freshness check passes on ubuntu-latest + Python 3.13 in CI
- [ ] Manually verify `--mosaic` flag works for HST ACS/WFC and WFPC2 images
- [ ] Verify `.LBL` label paths dispatch correctly for Cassini ISS and NH LORRI fixtures

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added `--mosaic` for multi-detector/panel assembly (replacing `--hst`) and updated related option validation and normalization.

* **Bug Fixes & Improvements**
  * Improved image input handling with a unified reader cascade and instrument-specific `read_file` flow, including optional `--tint`/`--mosaic` hook behavior.
  * Updated PDS3 label handling defaults to faster parsing.

* **Documentation**
  * Refreshed user/developer guides to match the new instrument protocol and `--mosaic` behavior.

* **API Updates**
  * Renamed 16-bit TIFF helpers to `read_tiff16` / `write_tiff16` (lowercase).

### CODE REVIEW

File: src/picmaker/io.py [hunk 1]
```diff
@@ -147,6 +117,30 @@ def read_one_image_array(
             :class:`ExceptionGroup` so callers can inspect what each
             reader rejected.
     """
+    # ---- Auto-parse .LBL paths ----
+    if isinstance(filename, (str, os.PathLike)):
+        path_str = str(filename)
+        if path_str.upper().endswith('.LBL'):
+            filename = pdsparser.PdsLabel(
+                path_str, method=kwargs.get('pds3_label_method', 'fast')
+            )
```
[the remark this answers is not in the document]
Comment (markshowalter) [reply src/picmaker/io.py]: No

File: tests/test_shared_branches.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        self._filepath = filepath
+        self._d = d
+
+    def as_dict(self) -> dict[str, Any]:
+        return self._d
+
+
+def test_is_fits_file_type_error_returns_false() -> None:
+    """Passing a non-path-like value (None) returns False without raising."""
+    assert is_fits_file(None) is False  # type: ignore[arg-type]
```
[the remark this answers is not in the document]
Comment (markshowalter) [reply tests/test_shared_branches.py:34]: Ignore

File: tests/test_shared_branches.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+class _FakeLabel:
+    """Minimal stand-in for pdsparser.PdsLabel used to unit-test _shared internals."""
+
+    def __init__(self, filepath: str, d: dict[str, Any]) -> None:
+        self._filepath = filepath
+        self._d = d
+
+    def as_dict(self) -> dict[str, Any]:
+        return self._d
```
[the remark this answers is not in the document]
Comment (markshowalter) [reply tests/test_shared_branches.py:29]: No
